src/proposals/Proposals.jl

```diff
@@ -138,5 +139,97 @@ function (prop::RWalk)(rng::AbstractRNG,
     return u, v, logl, ncall
 end
 
+"""
+    Proposals.RStagger(;ratio=0.5, walks=25, scale=1)
+
+Propose a new live point by random staggering away from an existing live point. 
+This differs from the random walk proposal in that the step size here is exponentially adjusted
+to reach a target acceptance rate _during_ each proposal, in addition to _between_
+proposals.
+
+`ratio` is the target acceptance ratio, `walks` is the minimum number of steps to take, and `scale` is the proposal distribution scale, which will update _between_ proposals.
+"""
+
+@with_kw mutable struct RStagger <: AbstractProposal
+    ratio = 0.5
+    walks = 25
+    scale = 1.0
+end
```

Comment:
I've added some assertion checks for the `RWalk` proposal that should be mimicked here-
```suggestion
    ratio = 0.5
    walks = 25
    scale = 1.0

    @assert 0 ≤ ratio ≤ 1 "Target acceptance ratio must be between 0 and 1"
    @assert walks > 0 "Number of steps must be positive"
    @assert scale ≥ 0 "Proposal scale must be non-negative"
end
```

for your info, these are not part of Julia's normal struct definition- these are part of the Parameters.jl interface